cms/djangoapps/contentstore/views/item.py

```diff
@@ -751,6 +762,35 @@ def _get_module_info(xblock, rewrite_static_links=True, include_ancestor_info=Fa
         return xblock_info
 
 
+def _get_gating_info(course, xblock):
+    """
+    Returns a dict containing gating information for the given xblock which
+    can be added to xblock info responses.
+
+    Arguments:
+        course (CourseDescriptor): The course
+        xblock (XBlock): The xblock
+
+    Returns:
+        dict: Gating information
+    """
+    info = {}
+    if xblock.category == 'sequential' and course.enable_subsection_gating:
+        info["is_prereq"] = gating_api.is_prerequisite(course.id, xblock.location)
+        info["prereqs"] = [
+            p for p in course.gating_prerequisites if unicode(xblock.location) not in p['namespace']
+        ]
+        prereq, prereq_min_score = gating_api.get_required_content(
+            course.id,
+            xblock.location
+        )
+        info["prereq"] = prereq
+        info["prereq_min_score"] = prereq_min_score
+        if prereq:
+            info["visibility_state"] = VisibilityState.staff_only
```

Comment:
A new visibility state should be created. Returning "staff_only" is very confusing (even if we wish to render them in similar ways).